module-2/backend/week_four.md

```diff
@@ -10,14 +10,26 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 ### Questions
 
 * What is a cookie?
+    * A Cookie is a data structure that is sent back and forth between client and server
 * What’s the difference between a session and a cookie?
+    * A session tracks a clients data from the server side where a cookie is exchanged between client and server
 * What’s a flash and when do you want to use flashes?
+    * Flashes notify the client when something needs to be communicated from the server
 * Why do people say “HTTP is stateless”?
+    * HTTP is stateless because it doesn't track any previous requests. It only sends and receives as it were the first time
 * What’s authentication? Explain.
+    * Authentication is the process of determining you are who you say you are. This can be done through password verification. 
 * What’s the difference between authentication and authorization?
+    * Authorization is determining what a client can do after they have been authenticated.(ie. Admins have more privileges than visitors)
 * What’s a before filter?
+    * before action specifies that a method must be run before any controller actions can be carried out. (ie. require_admin)
 * How do we keep track of a user once they’ve logged in?
+    * A user is tracked by a session hash once they are logged in
 * When do you want to namespace a resource? When do you want to nest a resource? What's the differences between those two approaches?
+    * Namespace is used when the parent resource isn't needing any controller actions
 * At a high level, what tools can you use to implement authorization? How would you use them?
+    * In order to implement authorization, enums, before actions, and namespacing can be used. 
 * What's an enum, and what advantages does it offer? What data type needs to be in your database to use an enum? Where do you declare an enum?
+    * An enum is used on a user model in order to determine what role they have. A default value needs to be declared in the database column.
```

Comment:
not ALWAYS A role but good answer. they the flexibility of a string, but the speed and efficiency of an integer